src/web/services/link.service.spec.ts

```diff
@@ -60,9 +60,9 @@ describe('Link Service', () => {
       .toBe('/web/generalkenobiyouareaboldone?instructorid=hello%20there');
   });
 
-  it('should generate the record page link', () => {
-    expect(service.generateRecordsPageLink(mockStudent, 'from my point of view the jedi are evil'))
-      .toBe(`${window.location.origin}/web/instructor/students/records?courseid=dog.gma-demo&studentemail=alice.`
+  it('should generate the student profile page link', () => {
+    expect(service.generateProfilePageLink(mockStudent, 'from my point of view the jedi are evil'))
+      .toBe('/web/instructor/courses/student/details?courseid=dog.gma-demo&studentemail=alice.'
```

Comment:
@samuelfangjw I have a question here. What's the purpose of `${window.location.origin}`? Why does some link have it in front but some not?

Reply:
Assuming this works as expected, this should be the difference between returning a link with absolute or relative path.